### PR TITLE
Fix power admin down up test

### PR DESCRIPTION
**Pull request by Ankur19**
* For every Fabric & Linecard reboot, validate Before & After Interfaces and their OperStatus.
* For ControllerCard reboot, wait until the primary supervisor becomes switchover-ready after power down up.

### CODE REVIEW

File: feature/platform/tests/power_admin_down_up_test/power_admin_down_up_test.go [hunk 1]
```diff
@@ -31,7 +31,38 @@ func TestFabricPowerAdmin(t *testing.T) {
...
+			after := intfOperStatus(t, dut)
+			for time.Since(start) < 3*time.Minute {
+				if len(after) == len(before) {
+					break
+				}
+				time.Sleep(30 * time.Second)
+				after = intfOperStatus(t, dut)
+			}
+			if len(after) != len(before) {
+				t.Errorf("Not all Interfaces came up after Power Down Up, got: %v, want: %v", after, before)
```
Comment (mojiiba) [inline feature/platform/tests/power_admin_down_up_test/power_admin_down_up_test.go:48]: this logging statement is not helpful when the test fails, can u please change it to use https://github.com/openconfig/featureprofiles/blob/132572aa0c1c210a57642717de4bf86c57903d51/internal/helpers/helpers.go#L57  to log the exact interfaces which are not up after powerup.

File: feature/platform/tests/power_admin_down_up_test/power_admin_down_up_test.go [hunk 2]
```diff
@@ -53,19 +84,51 @@ func TestLinecardPowerAdmin(t *testing.T) {
...
+				}
+				op, ok := before[intf.GetName()]
+				if !ok {
+					return false
+				}
+				return intf.GetOperStatus() == op
+			})
+
+			if _, ok := intfs.Await(t); !ok {
+				t.Errorf("Interface Oper-Status didn't retun to original state after PowerAdmin Down Up for %s.", l)
```
Comment (mojiiba) [inline feature/platform/tests/power_admin_down_up_test/power_admin_down_up_test.go:117]: same as above

File: feature/platform/tests/power_admin_down_up_test/power_admin_down_up_test.go [hunk 3]
```diff
@@ -31,7 +31,38 @@ func TestFabricPowerAdmin(t *testing.T) {
...
+				}
+				op, ok := before[intf.GetName()]
+				if !ok {
+					return false
+				}
+				return intf.GetOperStatus() == op
+			})
+
+			if _, ok := intfs.Await(t); !ok {
+				t.Errorf("Interface Oper-Status didn't retun to original state after PowerAdmin Down Up for %s.", f)
```
Comment (mojiiba) [inline feature/platform/tests/power_admin_down_up_test/power_admin_down_up_test.go:64]: this logging statement is not helpful when the test fails, can u please change it to use

[featureprofiles/internal/helpers/helpers.go](https://github.com/openconfig/featureprofiles/blob/132572aa0c1c210a57642717de4bf86c57903d51/internal/helpers/helpers.go#L57)

Line 57 in [132572a](https://github.com/openconfig/featureprofiles/commit/132572aa0c1c210a57642717de4bf86c57903d51)
 func ValidateOperStatusUPIntfs(t *testing.T, dut *ondatra.DUTDevice, upIntfs []string, timeout time.Duration) { 
to log the exact interfaces which are not up after powerup.

Comment (Ankur19) [reply feature/platform/tests/power_admin_down_up_test/power_admin_down_up_test.go]: Using the helper now! Thanks for the pointer

**Review by rszarecki — APPROVED** [review]

**Pull request merged**